internal/setup/setup.go

```diff
@@ -151,14 +173,16 @@ func (r *Release) validate() error {
 			for newPath, newInfo := range new.Contents {
 				if old, ok := paths[newPath]; ok {
 					oldInfo := old.Contents[newPath]
+					// Note that if extracting content (CopyPath or GlobPath)
+					// from the same package the content can never be in conflict.
```

Comment:
In the other instances that you flagged in this PR I agree completely that the comments were redundant, however I think we need some comments in this case. The problem is that I tried to write something short here so it is halfway between a good comment and a description of the code.

The code comparing globs and content is difficult to understand if you don't remember the premise that we are checking for conflicts without downloading packages, and we can only be sure that the content matches if it is coming from the same one because of that precise reason. I think it will benefit readability if we included a general comment that does not talk about the code but about what "validation" means. What about the following?
```go
// A conflict means that two slices attempt to extract different files or
// directories to the same location.
// Conflict validation is done without downloading packages which means, that
// if we are extracting content from different packages to the same location we
// cannot be sure that it will be the same. On the contrary, content extracted
// from the same package will never conflict because it is guaranteed to be the
// same.
// The above also means that generated content (e.g. text files, directories
// with make:true) will always conflict with extracted content, because we
// cannot validate that they are the same without downloading the package.
```
Thoughts? Is it too verbose?

Reply:
Discussed offline that comment is valid because it talks about the reasoning and not about the code.

Reply:
Yeah, the comment above looks nice, thanks.